Source/ACE/Entity/Player.cs

```diff
@@ -1558,7 +1563,37 @@ protected override void SendUpdatePosition()
             Session.Network.EnqueueSend(msg);
         }
 
-        public void HandleActionPutItemInContainer(ObjectGuid itemGuid, ObjectGuid containerGuid)
+        private void HandleUnequip(WorldObject item, uint location)
```

Comment:
Unequip is a fine name -- keep the prefix though HandleAction alerts the caller that this is known callable from outside of the players action queue (it will enqueue any action it needs to run safely).